File: implementations/nestjs/examples/src/account/application/integration-event/account-closed-integration-event.ts
// The Integration Event (a public contract) the Account BC exposes to external BCs.
export class AccountClosedIntegrationEventV1 {
  public readonly eventName = 'account.closed.v1' as const

  constructor(
    public readonly accountId: string,
    public readonly closedAt: string
  ) {}
}

File: implementations/fastapi/examples/src/account/application/integration_event/account_closed_integration_event.py
from __future__ import annotations

from dataclasses import dataclass
from typing import ClassVar


@dataclass(frozen=True)
class AccountClosedIntegrationEventV1:
    """An Integration Event (a public contract) the Account BC exposes to external BCs (Card, etc.)."""

    event_name: ClassVar[str] = "account.closed.v1"
    account_id: str
    closed_at: str
